### ISSUE TITLE
Command Line Interface

### ISSUE THREAD

**Issue by Meerkov** [vgmtrans/vgmtrans#297]
**Is your feature request related to a problem? Please describe.**
Hey, so I'm looking to use this from the command line. I've more or less got a series of hacks on top of the existing infrastructure to allow this, but it might be neat if there was a more official way to do it.

**Describe the solution you'd like**

- [ ] CLI Commands for loading a file and automatically exporting all discovered files
- [ ] File names based on running count, rather than everything named the same thing
- [ ] Exported files based on combination of the file names.
- [ ] Perform pairwise matches when there is more than one possible match for a VAB to a Sample Collection.
- [ ] Exporting needs to take an output directory, rather than use the filepicker.

**Additional context**
I think the primary first task is just to create a replacement for the WinForms root called CLIRoot. This root will simply dummy out all of the UI function calls.

Currently, in the WTL, there is command line arguments, but only for loading a file in the UI. Unhooking the UI from the parsing requires some additional steps, as does exporting.

**Comment by sykhro** [comment]
> I think the primary first task is just to create a replacement for the WinForms root called CLIRoot. This root will simply dummy out all of the UI function calls.

Yes, subclassing Root is mandatory as we don't really want to touch WTL anymore and the Qt UI wasn't built with command line usage in mind (there are some improvements wrt mass export that I want to add before the big stable release).
For future proofing and ease of maintainance, we have a few requirements:
- it has to work on all platforms we currently support
- it should live as a self-contained ui implementation under `src/ui/`
- ideally, it should avoid pulling external libraries besides the ones we already use

This would be a start. We can make changes to the detection/matching logic once we have an interface that mostly works.

**Comment by Meerkov** [comment]
It sounds like this is blocked on the Qt UI stable release.
